core/io/file_access_pack.cpp

```diff
@@ -123,15 +123,17 @@ PackedData::~PackedData() {
 
 //////////////////////////////////////////////////////////////////
 
-bool PackedSourcePCK::try_open_pack(const String &p_path, bool p_replace_files) {
+bool PackedSourcePCK::try_open_pack(const String &p_path, bool p_replace_files, int p_offset) {
 	FileAccess *f = FileAccess::open(p_path, FileAccess::READ);
 	if (!f) {
 		return false;
 	}
 
+    f->seek(p_offset);
+
 	uint32_t magic = f->get_32();
 
-	if (magic != PACK_HEADER_MAGIC) {
+	if (magic != PACK_HEADER_MAGIC && p_offset == 0) {
 		//maybe at the end.... self contained exe
 		f->seek_end();
 		f->seek(f->get_position() - 4);
```

Comment:
We suppose that load with offset feature won't be used together with [self contained exe feature](https://github.com/godotengine/godot/pull/24086). This allows us to only add one extra argument (but not size), giving us a cleaner interface.

Reply:
Added comment.